kong/runloop/plugins_iterator.lua

```diff
@@ -390,23 +422,63 @@ local function new_ws_data()
 end
 
 
+local function configure(configurable, ctx)
+  ctx = ctx or ngx.ctx
+  local kong_global = require "kong.global"
+  for _, plugin in ipairs(CONFIGURABLE_PLUGINS) do
+    local name = plugin.name
+
+    kong_global.set_namespaced_log(kong, plugin.name, ctx)
+    local start = utils.get_updated_monotonic_ms()
+    local ok, err = pcall(plugin.handler[CONFIGURE_PHASE], plugin.handler, configurable[name])
+    local elapsed = utils.get_updated_monotonic_ms() - start
+    kong_global.reset_log(kong, ctx)
+
+    if not ok then
+      kong.log.err("failed to execute plugin '", name, ":", CONFIGURE_PHASE, " (", err, ")")
+    else
+      if elapsed > 50 then
+        kong.log.notice("executing plugin '", name, ":", CONFIGURE_PHASE, " took excessively long: ", elapsed, " ms")
+      end
+    end
+  end
+end
+
+
+local function create_configure(configurable)
+  if kong.configuration.role == "control_plane"
+  or ((subsystem == "http"   and #kong.configuration.proxy_listeners == 0) or
+      (subsystem == "stream" and #kong.configuration.stream_listeners == 0))
+  then
```

Comment:
We have this check in multiple places. We may benefit from a generic function that does these checks

Reply:
I agree. Let's do it in other PR.